### ISSUE TITLE
doesnt support `encoding='utf-8'`

### ISSUE THREAD

**Issue by sandrotosi** [aklajnert/pytest-subprocess#25]
i have a function that executes:

```
def f():
    return subprocess.check_output('secret_command', encoding='utf-8').strip()
```

but when i test it with:

```
    retvalue = '1029384756'
    with fake_process.context() as nested_process:
        nested_process.register_subprocess('secret_command', stdout=retvalue)
        node_id = f()
        assert node_id == retvalue
```

it fails with:

```
>           assert node_id == retvalue
E           AssertionError: assert b'1029384756' == '1029384756'
```

i believe because `stdout` is passed to `_prepare_buffer()` https://github.com/aklajnert/pytest-subprocess/blob/master/pytest_subprocess/core.py#L135 which will encode() the output. there's a chance to avoid it by setting `self.text` but i couldnt find a way from `register_subprocess` to set that

**Comment by aklajnert** [comment]
The return value from the subprocess is byte-encoded - you can tell that by the `b''` prefix. You need to call `node_id.decode()` or use `text=True` argument for the `subprocess.check_output()` to compare the values.

```python
assert b'1029384756'.decode() == '1029384756'
```

**Comment by sandrotosi** [comment]
i used `encoding='utf-8'` which returns a unicode string in python3, not a byte object; that's what i'm referring to

**Comment by sandrotosi** [comment]
```
In [7]: subprocess.check_output('echo aasdf', shell=True)                                                                                                                                                                                                                       
Out[7]: b'aasdf\n'

In [8]: subprocess.check_output('echo aasdf', shell=True, encoding='utf-8')                                                                                                                                                                                                     
Out[8]: 'aasdf\n'
```

**Comment by aklajnert** [comment]
Oh, that's true. I didn't know about it. I'll fix it soon.

**Comment by aklajnert** [comment]
This has been fixed in #26 and released as version `0.1.5`.

**Comment by sandrotosi** [comment]
perfect, it works like a charm!
